pipeline/src/pipeline.rs

```diff
@@ -865,7 +866,10 @@ impl<T: FieldElement> Pipeline<T> {
         self.arguments.query_callback.as_deref()
     }
 
-    pub fn verification_key(&mut self) -> Result<Vec<u8>, Vec<String>> {
+    pub fn write_verification_key<W: io::Write>(
```

Comment:
Why does Pipeline write it to FS instead of just returning the key?

Reply:
Why is the pipeline returning a key in the first place?

I am very uncomfortable with uninterpreted `Vec<u8>` being passed around contained backend specific serialized data. We are bypassing the type system. This is no better than casting to `void *` or using `unsafe` to reinterpret cast. Serializations should go directly to disk.

Reply:
I disagree it should go directly to disk. I think the user should decide what to do with it. Vec<u8> is a universal serialized data type.

Reply:
The proposed API accommodates both `File` and `Vec<u8>`. Both implement `io::Write`.